Mathlib/Topology/Sheaves/Stalks.lean

```diff
@@ -88,46 +88,74 @@ theorem stalkFunctor_obj (ℱ : X.Presheaf C) (x : X) : (stalkFunctor C x).obj 
 
 /-- The germ of a section of a presheaf over an open at a point of that open.
 -/
-def germ (F : X.Presheaf C) {U : Opens X} (x : U) : F.obj (op U) ⟶ stalk F x :=
-  colimit.ι ((OpenNhds.inclusion x.1).op ⋙ F) (op ⟨U, x.2⟩)
+def germ (F : X.Presheaf C) (U : Opens X) (x : X) (hx : x ∈ U) : F.obj (op U) ⟶ stalk F x :=
+  colimit.ι ((OpenNhds.inclusion x).op ⋙ F) (op ⟨U, hx⟩)
```

Comment:
Sorry, I don't have a huge amount of intuition about general presheaves, but isn't there always a morphism `F U -> stalk F x`, regardless of whether x is in U or not? My mental picture is the presheaf of continuous functions, and there the stalk at a point contains eg constant functions. This would mean that you can use a junk value when x is not in U and therefore get rid of `hx` in the definition

Reply:
I don't think you can guarantee `F.obj (op U) ⟶ stalk F x` is nonempty (so that you can choose a junk map) for general categories. And even in `Type`, `stalk F x` might not be non-empty.